modules/apps/portal/portal-upgrade-impl/src/main/java/com/liferay/portal/upgrade/internal/report/UpgradeReport.java

```diff
@@ -45,10 +67,111 @@ public void addWarningMessage(String loggerName, String message) {
 		warningMessages.add(message);
 	}
 
+	public void generateReport() {
+		File file = new File(PropsValues.LIFERAY_HOME, "upgrade_report.info");
```

Comment:
@SamZiemer we need to add the datetime (using the same format than for the upgrade logs) to the file name. Having a specific folder for the upgrade reports would be perfect but we can implement it later if it is problematic.